pkg/model/datacatalog/api.go

```diff
@@ -53,3 +53,23 @@ type DeleteAssetResponse struct {
 	// The deletion status
 	Status string `json:"status,omitempty"`
 }
+
+type UpdateAssetRequest struct {
+	// The destination catalog id in which the asset will be updated based on the information provided
+	// in ResourceMetadata and ResourceDetails field
+	// Asset ID of the to-be updated asset
+	AssetID taxonomy.AssetID `json:"assetID"`
+	// Source asset metadata like asset name, owner, geography, etc
+	ResourceMetadata ResourceMetadata `json:"resourceMetadata"`
+	// Source asset details like connection and data format
+	Details ResourceDetails `json:"details"`
```

Comment:
can we update all asset parameters? I mean metadata (asset name, geography...) and asset details (connection, data format), my guess is that only a subset of parameters we can change. If we cannot change some of the parameters, maybe we don't need to include them in the request.

Reply:
Please suggest which fields can be updated. Based on that, we need to edit the  [UpdateAssetRequest] struct. (https://github.com/rohithdv/fybrik/blob/649819aef62ec7b51a8e23501c15653d746d566f/pkg/model/datacatalog/api.go#L57). I think from my point of view, we could potentially update any field but if there are some hard constraints that some fields cannot change, then we can select those fields and use them in the `UpdateAssetRequest` struct. @roytman @shlomitk1

Reply:
If details change, it is not the same asset any longer. Path to the credentials is unlikely to change either. Which leaves us with metadata only, specifically with dataset tags and column tags.

Reply:
I think we can change the HTTP method from PUT to PATCH, and remove `Details` and `Credentials` from the request. Therefore we have `AssetID` and `ResourceMetadata`. 
Now, in the `ResourceMetadata` there is `Geography` I don't think it can be changed. If we move the asset to another location, it will be another asset. 
I'd suggest flattering `ResourceMetadata` in the UpdateAssetRequest request:

```golang
type UpdateAssetRequest struct {
	// The id of the dataset to be updated
	AssetID taxonomy.AssetID `json:"assetID"`
	// New name of the resource
	Name string `json:"name,omitempty"`
	// New owner of the resource
	Owner string `json:"owner,omitempty"`
	// New tags associated with the asset
	Tags *taxonomy.Tags `json:"tags,omitempty"`
	// New columns associated with the asset
	Columns []ResourceColumn `json:"columns,omitempty"`
}
```
We can document that in the case of collection (tags and columns) updates, the update will be done for the entire collection. Fine grine updates (add, remove, update of s specific element) will be checked in the future releases.

Reply:
I have changed to PATCH as per the suggestions. Please review. Thanks. @roytman